### PR TITLE
csr validation test: compare aggregate error

### LINKED ISSUE #103549: TestValidateCertificateSigningRequestCreate/valid_duration is flaking
#### Which jobs are flaking:

pull-kubernetes-unit

#### Which test(s) are flaking:

pkg/apis/certificates/validation TestValidateCertificateSigningRequestCreate/valid_duration

#### Testgrid link:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/98817/pull-kubernetes-unit/1412827544935206912

#### Reason for failure:

```
    validation_test.go:378: returned and expected errors did not match - expected
        <nil>
        but got
        <nil>
```

cc @enj for #99494

### PR DESCRIPTION
**Pull request by enj**
Signed-off-by: Monis Khan <mok@vmware.com>

/kind failing-test
/kind flake
/sig auth
/priority critical-urgent

Fixes #103549 

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/certificates/validation/validation_test.go [hunk 1]
```diff
@@ -378,7 +378,7 @@ func TestValidateCertificateSigningRequestCreate(t *testing.T) {
 	for name, test := range tests {
 		t.Run(name, func(t *testing.T) {
 			el := ValidateCertificateSigningRequestCreate(&test.csr, test.gv)
-			if !reflect.DeepEqual(el, test.errs) {
```
Comment (alculquicondor) [inline pkg/apis/certificates/validation/validation_test.go:381]: Can you tell why this solves the issue?

also, `cmp.Diff` would probably give you more information for debugging than `reflect.DeepEqual`

Comment (alculquicondor) [reply pkg/apis/certificates/validation/validation_test.go:381]: I'm thinking that the problem is on `errs: field.ErrorList{}`. Probably better to use just leave it at `nil`

Comment (enj) [reply pkg/apis/certificates/validation/validation_test.go:381]: No, the validation methods return `field.ErrorList{}`, not `nil`.

Comment (alculquicondor) [reply pkg/apis/certificates/validation/validation_test.go:381]: can you replace the check with `cmp.Diff` that will give us information if we see still see the problem in the jobs.

**Comment by alculquicondor** [comment]
/hold

I think this is just failing for my PR because I changed the validation logic to return nil. I'll take it.

**Comment by liggitt** [comment]
yeah... all the failures are on your PR... sorry I didn't notice that sooner

/close

you can deal with the issue in your PR

**Pull request closed without merging**